libguard/guard_interface.cpp

```diff
@@ -234,127 +240,74 @@ GuardRecords getAll()
     return guardRecords;
 }
 
-/**
- * @brief Helper function to delete guard record
- *
- * @param[in] record to delete
- * @param[in] recordPos position for deleting record
- * @param[in] posOfLastRecord position of last record in guard file
- *
- * @return NULL on success
- *         Throw exception on failure
- *
- */
-static void deleteRecord(GuardRecord record, int recordPos, int posOfLastRecord)
-{
-    GuardRecord nullGuard;
-    size_t sizeOfGuardRecord = sizeof(nullGuard);
-
-    memset(&nullGuard, 0xFF, sizeOfGuardRecord);
-
-    uint32_t offset = recordPos * sizeOfGuardRecord;
-
-    GuardFile file(guardFilePath);
-    file.erase(offset + headerSize, sizeOfGuardRecord);
-
-    int i = recordPos + 1;
-    while (posOfLastRecord != i)
-    {
-        uint32_t offset1 = i * sizeOfGuardRecord;
-        file.read(offset1 + headerSize, &record, sizeOfGuardRecord);
-        uint32_t offset2 = (i - 1) * sizeOfGuardRecord;
-        file.write(offset2 + headerSize, &record, sizeOfGuardRecord);
-        file.write(offset1 + headerSize, &nullGuard, sizeOfGuardRecord);
-        i++;
-    }
-}
-
-void clear(const EntityPath& entityPath)
+static void invalidateRecord(const EntityPath& entityPath,
```

Comment:
This seems to error prone. you are assuming one of the value is good here. incase user provided both arguments then error path is an issue. Also not providing usefull debug data here for failure case.
Since no0 header in the function, am assuming you are expecting only one parameter at a time . to achive this you need to decalre as stad::variant and handle properly.

Otherwise use the existing flow in the code used for deleteRecord().

Reply:
using std::variant. Made the changes.

Reply:
if the record pos starts with 0 , then the logic is broken here . if that is the case intilize with FFFFFFFF. so that the logic holds good.

Reply:
No, record position i.e. recordId can never be 0. We start it with 1. If we initialize with FFFF..FF than it will try to invalidate the already invalidated guard record(because FFFFFFFF is a recordId in guard i.e. invalidated guard record)